### PR TITLE
Feature/remove depricated colors

### LINKED ISSUE #1748: [Child of #1755] Colors are outdated
Feedback from @MKaHead

Some colors are deprecated and should be removed. And the colors file should be split.

### PR DESCRIPTION
**Pull request by 01231**
Fixes #1748

# Done is when (DoD):
- [x] Modifications within components are reflected by tests.
- [x] A self-review of the code has been performed.
- [x] The modified component properties have been added to the typescript typings.
- [x] Potential design changes have been reviewed by a designer.
- [x] The modifications are well documented (README.md, etc.) and showcased in the stories.
- [x] The modifications are documented in the code, particularly in hard-to-understand areas (focus on **WHY**).
- [x] The modifications are shortly added to CHANGELOG.md with the issue number.
- [x] The modifications still work in IE.
- [x] The modifications are according to our [Best Practices](https://github.com/axa-ch/patterns-library/blob/develop/CONTRIBUTION.md#best-practices)

### CODE REVIEW

File: src/components/00-materials/styles/story.js [hunk 1]
```diff
@@ -139,7 +117,7 @@ storiesOf('Materials', module)
           width: 250px;
           height: 50px;
           border-radius: 5px;
-          border: 1px solid lightgray;
+          border: 1px solid lightgrey;
```
Comment (danyball) [inline src/components/00-materials/styles/story.js:120]: Isn't `lightgray` and `lightgrey` the same colorcode?

File: src/components/10-atoms/button-link/index.scss [hunk 2]
```diff
@@ -24,15 +24,15 @@
...
   }
 
   &[aria-disabled="true"] {
-    color: $color-prim-gray-dusty;
-    background-color: $color-prim-gray-mercury;
-    border-color: $color-prim-gray-mercury;
-    border-bottom-color: $color-prim-gray-silver;
+    color: $color-light-grey;
+    background-color: $colo-mercury;
+    border-color: $colo-mercury;
```
Comment (danyball) [inline src/components/10-atoms/button-link/index.scss]: `colo` typo

File: src/components/10-atoms/button/index.scss [hunk 3]
```diff
@@ -29,15 +29,15 @@
...
   }
 
   &:disabled {
-    color: $color-prim-gray-dusty;
-    background-color: $color-prim-gray-mercury;
-    border-color: $color-prim-gray-mercury;
-    border-bottom-color: $color-prim-gray-silver;
+    color: $color-light-grey;
+    background-color: $colo-mercury;
+    border-color: $colo-mercury;
```
Comment (danyball) [inline src/components/10-atoms/button/index.scss]: typo

File: src/components/20-molecules/datepicker/story.js [hunk 4]
```diff
@@ -55,7 +55,7 @@ story.add('Datepicker', () => {
   const wrapper = document.createElement('div');
 
   const template = html`
-    <div style="background-color: lightgray">
+    <div style="background-color: lightgrey">
```
Comment (danyball) [inline src/components/20-molecules/datepicker/story.js:58]: This `lightgrey` is the build-in css variable. Just to be sure you do not make a replace mistake. but i think its ok

**Review by danyball — CHANGES_REQUESTED** [review]
Nice!

File: src/components/00-materials/styles/story.js (hunk 1 above)
Comment (01231) [reply src/components/00-materials/styles/story.js:120]: yes but I decide to name everything "grey" for consistency.

File: src/components/20-molecules/datepicker/story.js (hunk 4 above)
Comment (01231) [reply src/components/20-molecules/datepicker/story.js:58]: This was on purpose. Shouldn't make a difference, due the fact that both have the same color code.

File: src/components/00-materials/styles/story.js (hunk 1 above)
Comment (LucaMele) [reply src/components/00-materials/styles/story.js:120]: american vs english grey

**Review by LucaMele — APPROVED** [review]
LGTM

**Review by danyball — APPROVED** [review]

**Pull request merged**